src/main/java/digital/slovensko/autogram/ui/gui/MainMenuController.java

```diff
@@ -38,21 +41,37 @@ public void initialize() {
         });
 
         dropZone.setOnDragDropped(event -> {
-            for (File file : event.getDragboard().getFiles()) {
-                SigningJob job = SigningJob.buildFromFile(file, new SaveFileResponder(file, autogram), false);
-                autogram.sign(job);
-            }
+            processFileList(event.getDragboard().getFiles());
         });
     }
 
     public void onUploadButtonAction() {
         var chooser = new FileChooser();
         var list = chooser.showOpenMultipleDialog(new Stage());
+        processFileList(list);
+    }
 
+    public void processFileList(List<File> list) {
         if (list != null) {
-            for (File file : list) {
-                SigningJob job = SigningJob.buildFromFile(file, new SaveFileResponder(file, autogram), false);
+            // If single directory is selected (dropped) we can process it as a batch
+            if (list.size() == 1) {
+                var file = list.get(0);
+                if (file.isDirectory()) {
+                    list = Stream.of(file.listFiles()).filter(f -> f.isFile()).toList();
+                }
```

Comment:
Mne sa to pri nahraní directory správa akokeby som nahral všetky súbory z neho - vo vnútri tohto sirectory sa vytvorí `signed` a do vnútra sa dajú popdísané súbory.
```
source_dir/
--- signed/
    --- file1_signed.asice
    --- file2_signed.asice
--- file1.idk
--- file2.idk
```

Nemalo by to byť takto?
```
source_dir/
--- file1.idk
--- file2.idk
signed/
--- file1_signed.asice
--- file2_signed.asice
```
Možno je to ale špecialita OS

Reply:
nie, toto je hack ak dropnes folder tak zoberiem vsetky subory v nom ako keby si ich selectol - mozno to je divne
ale keby som sa k tomu staval ako ku folderu tak by to bolo `source_dir_signed` 

(btw robil si to cez drag&drop? lebo mne sa neda vybrat folder cez picker a mozno linux nema to obmedzenie)

Reply:
ja som vybral source_dir cez filepicker, takže asi sa to na Gnome Ubuntu dá.

Reply:
ok, lebo som bol v tom, ze `FileChooser.showOpenMultipleDialog` ma obmedzenie na iba subory preto ten hack

co je podla Teba najlepsie spravanie? ked mam jeden folder, ked mam viacero folderov, ked mam zmiesane files & folders?

Reply:
Ja som čakal, že sa to bude správať ako pri CLI, čiže `source_dir_signed`. Tie zmiešané a vnorené veci sme zatiaľ odkladali, ale môžem si teda posnívať. Ja si predstavujem, že ak je vo výbere všeobecne viac ako jedna vec, tak to má vytvoriť `signed/` a do vnútra uložiť podpísané veci v rovnakej štruktúre ako boli na vstupe. (Dokonca si myslím, že do vnútra `signed/` adresára by sme už nemali prdávať sufix `_signed`) Ak existujú vnorené adresáre a súbory, spracoval by som všetko a v rovnakej štruktúre to ukladal do `signed/`. Rekurzívne by som ale default nerobil - mal by to user explicitne zapnúť. By default by som podpísal iba označené súbory/súbory priamo vo vybranom adresári.

Reply:
mame jemny problem - javafx nepodporuje  vyberanie suborov alebo priecinkov (vid https://stackoverflow.com/a/18237547 ) a na macos je to implementovane tak, ze sa neda zvolit priecinok ked vyberas subory

(a teraz uz asi 10-krat prepisujem to podpisovanie s priecinkami :D)
zatial som to upravil tak, ze by to nemalo vadit, nepodporujeme podpisovanie viacerych priecinkov a priecinky sa podpisuju iba ak su v tom originalnom vybrani (vtedy sa daju so _signed)

Reply:
fakty
- nemame vymysleny recursive mode
- naming schema kde ku folderom pridavame `_signed` (CLI) sposobuje, ze user ocakava folder1_signed, folder2_signed ked vyberie viacero folderov
- na linuxe sa vo FileChooser daju vybrat aj directories
- na macos sa daju drag&dropnut foldre a directories (aj naraz)
- Autogram class zarucuje, ze existuje iba jeden batch
- nevieme to dat do jedneho batchu kvoli dialogom ktore su pre jednotlivy batch ale my by sme potrebovali vypisat ze subory su ulozene na roznych miestach

Reply:
Tiez by som cakal takmer identicke spravanie ako pri CLI mode, a urcite by som sa teraz obmedzil na to, ze ked "miesas foldre a subory" tak to povie, ze nie. Dokonca si myslim, ze cez GUI bude bezny scenar viac suborov a nie "vsetko v tomto foldri" cize ak to chceme zjednodusit, tak spravme to, ze uplne vyrazme z GUI modu podporu pre foldre.